src/raft.c

```diff
@@ -341,23 +362,39 @@ void RaftExecuteCommandArray(RedisRaftCtx *rr,
         int old_entered_eval = rr->entered_eval;
 
         if ((cmdlen == 4 && !strncasecmp(cmd, "eval", 4)) ||
-            (cmdlen == 7 && !strncasecmp(cmd, "evalsha", 7))) {
+            (cmdlen == 7 && !strncasecmp(cmd, "evalsha", 7)) ||
+            (cmdlen == 5 && !strncasecmp(cmd, "fcall", 5))) {
             rr->entered_eval = 1;
         }
 
+        char *resp_call_fmt;
```

Comment:
```c
        char *resp_call_fmt = cmds->acl ? "CE0v" : "E0v";

        enterRedisModuleCall();
        RedisModule_SetContextUser(ctx, user);
        reply = RedisModule_Call(ctx, cmd, resp_call_fmt, &c->argv[1], c->argc - 1);
        exitRedisModuleCall();

        rr->entered_eval = old_entered_eval;

        if (reply_ctx) {
            RedisModule_ReplyWithCallReply(reply_ctx, reply);
        }

	if (reply) {
            RedisModule_FreeCallReply(reply);
        }
```
Starting from this line and to the end, can we do this? 

Note: I think crashes you see are because you reply to the `ctx`, not to `reply_ctx`. `reply_ctx` occasionally will be NULL.

Reply:
less allocations on followers.  so "need" is relative.

Reply:
Yes I see but it only helps on error cases (I assume error cases are rare). 

My only concern is readability. Flags are already hard to read and explain :) so, at least we can skip this extra check as it is not going to help us anyway.

Reply:
I still don't understand why redis went with a "flags" as characters method vs binary bits with #defines.  I get it for denoting the arguments, but not for the non arguments.

Reply:
Yes, it is a bit counter-intuitive.

Reply:
Yes, format string gets uglier this way when you start adding new flags.

Reply:
Yes, format string gets uglier this way when you start adding new flags.